src/gravel/controllers/nodes/mgr.py

```diff
@@ -467,6 +431,31 @@ def _load_token(self) -> Optional[str]:
         token = TokenModel.parse_file(tokenfile)
         return token.token
 
+    def _save_token(self, should_exist: bool = True) -> None:
+        tokenfile: Path = self._get_node_file("token")
+
+        # this check could be a single assert, but it's important to know which
+        # path failed.
+        if should_exist:
+            assert tokenfile.exists()
+        else:
+            assert not tokenfile.exists()
```

Comment:
I'm confused, do we really need to assert this?

Reply:
We don't need to assert it, but it sure comes handy to ensure we are getting the right state at the right time from the caller's PoV. The alternative would be having the caller perform the asserts first, but that would require them to know where the file sits and hence increase boilerplate quite a bit.